frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -978,6 +1003,64 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        // Set Delegate Rewards to an account
+        #[pallet::weight(T::DbWeight::get().writes(1))]
+        pub fn set_delegate_rewards(
+            origin: OriginFor<T>,
+            delegate_account: T::AccountId,
+            contract_id: T::SmartContract,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            let staker = ensure_signed(origin)?;
+
+            // check origin is different from delegate account
+            ensure!(
+                staker != delegate_account,
+                Error::<T>::DelegateRewardAccountIsStaker
+            );
+
+            if !DelegateRewardAccounts::<T>::contains_key(&staker, &contract_id) {
+                DelegateRewardAccounts::<T>::insert(
+                    &staker,
+                    &contract_id,
+                    delegate_account.clone(),
+                );
+            } else {
+                DelegateRewardAccounts::<T>::mutate(&staker, &contract_id, |delegate_account| {
+                    *delegate_account = delegate_account.clone();
+                });
+            }
```

Comment:
How many reads/writes does each _arm_ here adds?

Can this be optimized?

Reply:
I think i improve the extrincit . Now if the record already exist it throw you an error and if not its insert the storage.